test/scenarios.py

```diff
@@ -5198,4 +5304,524 @@
+                            FILES["testfile-big"].id,
+                            "/tmp/received/29-2/testfile-big",
+                        )
+                    ),
+                    action.CheckDownloadedFiles(
+                        [
+                            action.File("/tmp/received/29-2/testfile-big", 10485760),
+                        ],
+                    ),
+                    action.CancelTransferRequest(0),
+                    action.ExpectCancel([0], False),
+                    action.NoEvent(),
+                    action.Stop(),
+                ]
+            ),
+        },
+    ),
+    Scenario(
+        "scenario29-3",
+        "Send three files to a peer, download one, reject one and do nothing with third one. Then stop the sender and then start back. Expect automatically restored transfer, without the rejected and stopped file",
+        {
+            "ren": ActionList(
+                [
+                    action.ConfigureNetwork(),
+                    action.Start("172.20.0.5", dbpath="/tmp/db/29-3-ren.sqlite"),
+                    action.WaitForAnotherPeer(),
+                    action.NewTransfer(
+                        "172.20.0.15",
+                        [
+                            "/tmp/testfile-big",
+                            "/tmp/testfile-bulk-01",
+                            "/tmp/testfile-bulk-02",
+                        ],
+                    ),
+                    action.Wait(
+                        event.Queued(
+                            0,
+                            {
+                                event.File(
+                                    FILES["testfile-big"].id,
+                                    "testfile-big",
+                                    10485760,
+                                ),
+                                event.File(
+                                    FILES["testfile-bulk-01"].id,
+                                    "testfile-bulk-01",
+                                    10485760,
+                                ),
+                                event.File(
+                                    FILES["testfile-bulk-02"].id,
+                                    "testfile-bulk-02",
+                                    10485760,
+                                ),
+                            },
+                        )
+                    ),
+                    action.Wait(
+                        event.FinishFileRejected(0, FILES["testfile-bulk-01"].id, True)
+                    ),
+                    action.Wait(event.Start(0, FILES["testfile-big"].id)),
+                    # wait for the initial progress indicating that we start from the beginning
+                    action.Wait(event.Progress(0, FILES["testfile-big"].id, 0)),
+                    # make sure we have received something, so that we have non-empty tmp file
+                    action.Wait(event.Progress(0, FILES["testfile-big"].id)),
```

Comment:
These three might racy, maybe wrap them inside Action.WaitRacy

Reply:
Actually, it's not racy. If the order is different then it's a manifestation of a bug

Reply:
Could you elaborate? Isn't that each file is being asynchronously started? Does this mean the transfers are initiated one by one?

Reply:
It is started only if it was in flight during the disconnection. In this test case, we only issue download for one file so the events are in order

Reply:
Oh I think I got confused because it's two progress events for the same file